File: autostart.py
"""开机自启：读写注册表 HKCU\\...\\Run 下的 TaskbarLyrics 项。"""
import os
import sys
import winreg

_RUN_KEY = r"Software\Microsoft\Windows\CurrentVersion\Run"
_VALUE_NAME = "TaskbarLyrics"


def _command() -> str:
    if getattr(sys, "frozen", False):
        return f'"{sys.executable}"'
    # 开发模式：用 venv 的 pythonw 启动 main.py
    root = os.path.dirname(os.path.abspath(__file__))
    pythonw = os.path.join(root, ".venv", "Scripts", "pythonw.exe")
    return f'"{pythonw}" "{os.path.join(root, "main.py")}"'


def is_enabled() -> bool:
    try:
        with winreg.OpenKey(winreg.HKEY_CURRENT_USER, _RUN_KEY) as key:
            winreg.QueryValueEx(key, _VALUE_NAME)
        return True
    except OSError:
        return False


def set_enabled(enabled: bool) -> None:
    with winreg.CreateKey(winreg.HKEY_CURRENT_USER, _RUN_KEY) as key:
        if enabled:
            winreg.SetValueEx(key, _VALUE_NAME, 0, winreg.REG_SZ, _command())
        else:
            try:
                winreg.DeleteValue(key, _VALUE_NAME)
            except OSError:
                pass

File: wpf/TaskbarLyrics/Autostart.cs
// 开机自启：读写注册表 HKCU\...\Run 下的 TaskbarLyrics 项（移植自 autostart.py）。
using Microsoft.Win32;

namespace TaskbarLyrics;

internal static class Autostart
{
    private const string RunKey = @"Software\Microsoft\Windows\CurrentVersion\Run";
    private const string ValueName = "TaskbarLyrics";

    public static bool IsEnabled()
    {
        try
        {
            using var key = Registry.CurrentUser.OpenSubKey(RunKey);
            return key?.GetValue(ValueName) != null;
        }
        catch
        {
            return false;
        }
    }

    public static void SetEnabled(bool enabled)
    {
        using var key = Registry.CurrentUser.CreateSubKey(RunKey);
        if (enabled)
        {
            var exe = Environment.ProcessPath
                      ?? System.Diagnostics.Process.GetCurrentProcess().MainModule?.FileName
                      ?? "";
            key.SetValue(ValueName, $"\"{exe}\"");
        }
        else
        {
            try { key.DeleteValue(ValueName); } catch { /* 不存在时忽略 */ }
        }
    }
}
